### ISSUE TITLE
[Surv] Does not apply green shrapnel bomb when Serpent Sting expires

### ISSUE THREAD

**Issue by geraint425** [Hekili/hekili#1085]
**BEFORE SUBMITTING AN ISSUE:**

1.  build: v9.0.5-1.0.8

2.  Check to see if someone else has already opened an issue report here.  Please don't submit repeat reports.

**Describe the Bug/Issue**
When the buff expires on one target, or several, the addon does not offer to use it when there is an active green shrapnel bomb. And it also works very badly when targets have 4+ buffs from Serpent Sting, and we have a ready-made green shrapnel bomb, we in AOE do not extend the buff and lose it while the addon is waiting for the bombs to roll back. Now in rotation, surv has a lot of problems with the addon.

**To Reproduce**
- Go to '...'
- Click on '....'
- Scroll down to '....'
- See error

**Expected behavior**
If there is a Serpent Sting buff on targets, and its end is 2-3 seconds, and we have an active green shrapnel bomb, then we apply it. Likewise in AOE, if Serpent Sting ends on one of the many targets, then we use a green shrapnel bomb. In other other cases, we try to update the buff through the Serpent Sting.

**Screenshots**

**WHEN SUBMITTING AN ISSUE:**

Please provide the following information.  Leaving it out may result in your ticket being ignored (because I won't have the information needed to explore your report).

1.  The information from the Issue Reporting tab ( /hekili > Issue Reporting ).  This provides your talent and gear information.  You can paste it to pastebin.com and provide a link here.

**https://pastebin.com/XxUufFp2**

2.  If reporting a disagreement with the addon's recommendations, please take a snapshot when you observe the issue and provide the snapshot.  This is not a screenshot.  A snapshot can be generated by binding the Pause feature on the Toggles tab.  When you press Pause, a snapshot of how the addon decided on its CURRENT recommendations is generated and saved to the Snapshots tab.  Copy and paste this to pastebin.com and provide the link here.
 
**https://pastebin.com/tFVr3xZj**

3.  If reporting an error (i.e., the addon freezes and/or disables itself), please install the BugSack and BugGrabber addons as they will enable you to paste the exact error message, with detail that will help me identify where the code error is located.  You can paste the error information here or use pastebin.com for that as well.

**PUT ERROR MESSAGE(S) HERE:**

**Comment by Hekili** [comment]
The addon does not know the duration of the Serpent Sting effects on the other targets.  It only knows that it's there or it isn't.

**Comment by geraint425** [comment]
Is it possible how to do tracking? Because if we use Serpent Sting but not shrapnel bomb. This is a loss of dps. Since if we have a green shrapnel bomb then we use it to update the buff from Serpent Sting.

**Comment by Hekili** [comment]
The solution will probably not be to track duration on other targets.  What have you tried in your sims?

**Comment by geraint425** [comment]
In sims, I didn't do anything. I basically only tried it in the game itself and looked at the efficiency in Details. 
Well, I just followed the tips and guides that are described on icy-veins. 
I can, in principle, attach my [sim](https://www.raidbots.com/simbot/report/6rvptofmCM1SzgYMg5eZBy) where the battle goes with 5 goals. I looked at the rotation this is exactly what I described to you. That we try to hit the sharpnel bomb as often as possible. And use Serpent Sting on the maximum number of targets, so that in the next when we have a green sharpnel bomb we can hit the targets with strong AOE damage.
